google/resource_compute_instance.go

```diff
@@ -480,6 +497,15 @@ func resourceComputeInstanceCreate(d *schema.ResourceData, meta interface{}) err
 		disks = append(disks, bootDisk)
 	}
 
+	scratchDisks := []*compute.AttachedDisk{}
```

Comment:
Same as https://github.com/terraform-providers/terraform-provider-google/pull/122#discussion_r122057157, we can keep this element in block scope instead of function scope.

That would mean changing the one other use of `scratchDisks` from

```go
if len(scratchDisks) > 0 {
```

to

```go
if d.Get("scratch_disk.#").(int) > 0
```

which looks a bit grosser, but holds better semantics to me - we are giving a schema error by comparing schema elements.